java/ruleXtract/src/it/scala/uk/ac/cam/eng/PipelineTest.scala

```diff
@@ -0,0 +1,188 @@
+/**
+ * *****************************************************************************
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use these files except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * *****************************************************************************
+ */
+
+package uk.ac.cam.eng
+
+import resource._
+import scala.concurrent._
+import scala.concurrent.duration._
+import ExecutionContext.Implicits.global
+
+import scala.collection.JavaConversions.iterableAsScalaIterable
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.FileSystem
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.hbase.io.hfile.CacheConfig
+import org.apache.hadoop.hbase.io.hfile.HFile
+import org.apache.hadoop.io.IntWritable
+import org.apache.hadoop.io.SequenceFile
+import org.apache.hadoop.io.Writable
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
+import org.apache.hadoop.mapreduce.lib.input.MultipleInputs
+import org.apache.hadoop.mapreduce.lib.input.SequenceFileInputFormat
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat
+import org.apache.hadoop.util.ReflectionUtils
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.ConfigMap
+import org.scalatest.FunSuite
+import uk.ac.cam.eng.extraction.Rule
+import uk.ac.cam.eng.extraction.RuleExtractorTest
+import uk.ac.cam.eng.extraction.RuleString
+import uk.ac.cam.eng.extraction.hadoop.extraction.ExtractorJob
+import uk.ac.cam.eng.extraction.hadoop.features.phrase.Source2TargetJob
+import uk.ac.cam.eng.extraction.hadoop.features.phrase.Target2SourceJob
+import uk.ac.cam.eng.extraction.hadoop.merge.MergeJob
+import uk.ac.cam.eng.extraction.hadoop.merge.MergeJob.MergeFeatureMapper
+import uk.ac.cam.eng.extraction.hadoop.merge.MergeJob.MergeRuleMapper
+import uk.ac.cam.eng.rule.features.Feature
+import uk.ac.cam.eng.rule.retrieval.HFileRuleReader
+import uk.ac.cam.eng.util.CLI
+import uk.ac.cam.eng.util.CLI.FilterParams
+import uk.ac.cam.eng.extraction.hadoop.datatypes.FeatureMap
+import uk.ac.cam.eng.extraction.hadoop.datatypes.ExtractedData
+
+class PipelineTest extends FunSuite with BeforeAndAfterAll {
```

Comment:
Please document classes in some detail. What is PipelineTest supposed to do? Is this testing the full pipeline?